### ISSUE TITLE
Authorization to list a query / mutation in schema

### ISSUE THREAD

**Issue by alanwillms** [rebing/graphql-laravel#96]
Currently we can implement an `authorize` method in order to check if the authenticated user can _execute _a query or mutation, but even if the user is not able to run it, it will be listed in the schema documentation.

It would be nice if I could specify if a query and/or mutation can appear in the schema documentation.

**Comment by alanwillms** [comment]
This is our problem: we want to have only one end-point for GraphQL: `/graphql`:

* if the user is not authenticated, we will only show the `createToken` mutation
* if the user is an administrator, we will list all queries and mutations
* if the user is an analyst, we will list all queries
* if the user is a seller, we will only list mutations/queries related to sales
* etc.

**Comment by rebing** [comment]
Seems like dynamic schema generation is a controversial issue. There are valid thoughts from both sides here: https://github.com/graphql/graphql-js/issues/113

I think a way for implementing it would be to add a boolean `limit_introspection_by_authorize` option to config that would remove unauthorized Fields from schema, when enabled. Not sure how easy it is to implement without performance issues, but you can give it a try.

**Comment by alanwillms** [comment]
I've opened a PR (#101) so we can discuss it. The implementation suggestion I submitted is working and based on the way `graphql-ruby` works: https://github.com/rmosolgo/graphql-ruby/blob/master/lib/graphql/schema/warden.rb

There are no working unit tests in this repository, so I tested it against an application I work with. After we finish discussing the implementation details and we are happy with the code, I will write unit tests for it.
